### PR TITLE
Use fedora as base for image builder

### PR DESCRIPTION
**Pull request by kevinrizza**
Remove dependencies on Openshift and DockerHub for image build steps

### CODE REVIEW

File: upstream-builder.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16-alpine as builder
+FROM quay.io/fedora/fedora:34-x86_64 as builder
```
Comment (njhale) [inline upstream-builder.Dockerfile:1]: Was there a reason to change this from `alpine`? 

I suspect this image build is also broken now due to the `apk` calls below.

Comment (njhale) [inline upstream-builder.Dockerfile:1]: Can we rename this `Dockerfile` and drop the original?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
+FROM quay.io/fedora/fedora:34-x86_64 AS builder
```
Comment (njhale) [inline Dockerfile:2]: There are still a lot of openshift specifics in this file. I'm wondering if we can drop it entirely.

**Review by njhale — CHANGES_REQUESTED** [review]

**Comment by joelanford** [comment]
@kevinrizza can this PR be closed out now that we're building via goreleaser and distroless?

**Pull request closed without merging**